conda/plugins/types.py

```diff
@@ -214,3 +214,26 @@ class CondaSetting:
     description: str
     parameter: Parameter
     aliases: tuple[str, ...] = tuple()
+
+
+@dataclass
+class CondaHttpHeader:
+    """
+    Define vendor specific headers to include HTTP requests
+
+    For details on how this is used, see
+    :meth:`~conda.plugins.hookspec.CondaSpecs.conda_http_headers`.
+
+    :param name: name of the header (internal reference only)
+    :param description: description of the HTTP header and its purpose
+    :param header_name: name of the header used in the HTTP request
+    :param header_value: value of the header used in the HTTP request
+    :param header_hosts: host(s) for which this header should be used with; when not set the header
+                        will be included in all HTTP requests.
+    """
+
+    name: str
+    description: str
+    header_name: str
+    header_value: str
+    header_hosts: set[str] | None = None
```

Comment:
I think it's not needed to have a name and a header name field for this domain object, since internally they are both represented by strings it's useless duplication.

```suggestion
@dataclass
class CondaRequestHeader::
    """
    Define vendor specific headers to include HTTP requests

    For details on how this is used, see
    :meth:`~conda.plugins.hookspec.CondaSpecs.conda_request_headers`.

    :param name: name of the header used in the HTTP request
    :param description: description of the HTTP header and its purpose
    :param value: value of the header used in the HTTP request
    :param hosts: host(s) for which this header should be used with; when not set the header
                        will be included in all HTTP requests.
    """

    name: str
    description: str
    value: str
    hosts: set[str] | None = None
```